protos/flyteidl/plugins/sagemaker/trainingjob.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package flyteidl.plugins.sagemaker;
+
+option go_package = "github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins";
+
+
+message AlgorithmSpecification {
+    string training_image = 1;  // TODO: will this change output? should it be an input instead of a part of the task spec?
+    string training_input_mode = 2;
+    string algorithm_name = 3;   // TODO: will this change output? should it be an input instead of a part of the task spec?
```

Comment:
is this like the HPO algorithm? or the training algorithm. If its training algorithm it will change the output right?

Reply:
This is the training algorithm yes, and changing It will change the output.

Reply:
But I am not entirely sure how I should design the user interface (in SDK) for this.. Let me check the pytorch task